cairo/ethereum/cancun/vm/interpreter.cairo

```diff
@@ -320,3 +346,215 @@ func _execute_code{
+        let state = env.value.state;
+        let is_empty = account_exists_and_is_empty{state=state}(
+            [message.value.target.value.address]
+        );
+
+        if (is_empty.value != FALSE) {
+            // Add to touched accounts
+            tempvar x = 1;
+            let dict_ptr = cast(evm.value.touched_accounts.value.dict_ptr, DictAccess*);
+            hashdict_write{poseidon_ptr=poseidon_ptr, dict_ptr=dict_ptr}(
+                1, message.value.target.value.address, 1
+            );
+            tempvar new_touched_accounts = SetAddress(
+                new SetAddressStruct(
+                    dict_ptr_start=evm.value.touched_accounts.value.dict_ptr_start,
+                    dict_ptr=cast(dict_ptr, SetAddressDictAccess*),
+                ),
+            );
+
+            EvmImpl.set_touched_accounts{evm=evm}(new_touched_accounts);
+            tempvar range_check_ptr = range_check_ptr;
+            tempvar bitwise_ptr = bitwise_ptr;
+            tempvar keccak_ptr = keccak_ptr;
+            tempvar poseidon_ptr = poseidon_ptr;
+            tempvar env = env;
+            tempvar evm = evm;
+        } else {
+            tempvar x = 2;
+            tempvar range_check_ptr = range_check_ptr;
+            tempvar bitwise_ptr = bitwise_ptr;
+            tempvar keccak_ptr = keccak_ptr;
+            tempvar poseidon_ptr = poseidon_ptr;
+            tempvar env = env;
+            tempvar evm = evm;
+        }
+        EnvImpl.set_state{env=env}(state);
+        EvmImpl.set_env{evm=evm}(env);
+        let env = evm.value.env;
+        tempvar range_check_ptr = range_check_ptr;
+        tempvar bitwise_ptr = bitwise_ptr;
+        tempvar keccak_ptr = keccak_ptr;
+        tempvar poseidon_ptr = poseidon_ptr;
+        tempvar env = env;
+        tempvar evm = evm;
+    }
+    let range_check_ptr = [ap - 6];
+    let bitwise_ptr = cast([ap - 5], BitwiseBuiltin*);
+    let keccak_ptr = cast([ap - 4], KeccakBuiltin*);
+    let poseidon_ptr = cast([ap - 3], PoseidonBuiltin*);
+    let env_struct = cast([ap - 2], EnvironmentStruct*);
+    let evm_struct = cast([ap - 1], EvmStruct*);
+    tempvar env = Environment(env_struct);
+    tempvar evm = Evm(evm_struct);
+
+    // Prepare return values based on error state
+    if (cast(evm.value.error, felt) != 0) {
+        let msg = create_empty_message_call_output(evm.value.error);
+        drop_evm(evm);
+        tempvar msg = msg;
+        tempvar err = cast(0, EthereumException*);
+        return (msg, err);
+    }
+
+    assert [range_check_ptr] = evm.value.refund_counter;
+    let range_check_ptr = range_check_ptr + 1;
+    tempvar msg = MessageCallOutput(
+        new MessageCallOutputStruct(
+            gas_left=evm.value.gas_left,
+            refund_counter=U256(new U256Struct(evm.value.refund_counter, 0)),
+            logs=evm.value.logs,
+            accounts_to_delete=evm.value.accounts_to_delete,
+            touched_accounts=evm.value.touched_accounts,
```

Comment:
you are using `accounts_to_delete` and `touched_accounts`, segments that are squashed in `drop_evm `

It would be better to `let squashed_evm = squash_evm` (or `finalize_evm`) before doing this, and them using the squashed segments `squashed_evm.value.accounts.to_delete`